mipengine/controller/node_landscape_aggregator.py

```diff
@@ -199,47 +210,63 @@ def start(self):
     def stop(self):
         self.keep_updating = False
 
+    def set_registries(self, nodes, data_models, dataset_locations):
+        _node_registry = NodeRegistry(self._logger)
+        _node_registry.nodes = nodes
```

Comment:
There is also a problem with the `log_node_changes` if the `set_nodes` is not used in the `NodeRegistry`.

You would need to call the `log_node_changes` inside this method.